tools/analyzers/shadowpredecl/analyzer.go

```diff
@@ -0,0 +1,103 @@
+package shadowpredecl
+
+import (
+	"errors"
+	"go/ast"
+	"go/token"
+
+	"golang.org/x/tools/go/analysis"
+	"golang.org/x/tools/go/analysis/passes/inspect"
+	"golang.org/x/tools/go/ast/inspector"
+)
+
+// Doc explaining the tool.
+const Doc = "Tool to detect declarations that shadow predeclared identifiers by having the same name."
+
+const messageTemplate = "%s '%s' shadows a predeclared identifier with the same name. Choose another name."
+
+var predeclared = []string{"true", "false", "iota", "nil", "append", "cap", "close", "complex" /*"copy",*/, "delete", "imag",
```

Comment:
I commented out `copy` because it is a common function name pattern. However only non-exported functions will conflict with the built-in identifier, so maybe let's just rename such occurrences in our codebase?